File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        HashMap<Integer,Boolean> map = new HashMap<>();
        for(int val:arr){
            map.put(val,true);
        }
        
        int count = 0;
        int ans = 0;
        int size = arr[arr.length-1]+k;
        for(int i=1; i<=size; i++){
            if(map.get(i)==null){
              count++;
            }
            
            if(count==k) return i;
        }
        return count;
    }
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    for(const e of arr){
        if(k>=e) k++;
        else return k;
    }
    return k;
};